api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -428,6 +428,21 @@ model NodePoolProperties {
   spec: NodePoolSpec;
 }
 
+/** label is the k8s label type used for keys, taints and values on the nodepool nodes */
+@maxLength(63)
+@pattern("^[A-Za-z0-9/-._~%!$&'()*+,;=:]+$")
+scalar label extends string;
```

Comment:
I think the naming this type _label_ might be a bit confusing. It is not a label but a _label value_.
A label is a label_key/label_value pair . This is represented by Record<label_value>

Reply:
They also have different restrictions. For example a taint key has up to 253 characters whereas a taint value has up to 63 characters

Reply:
updated to reflect this